sdk/lib/opentelemetry/sdk/trace/span_data.rb

```diff
@@ -48,6 +48,17 @@ def hex_trace_id
                               def hex_parent_span_id
                                 parent_span_id.unpack1('H*')
                               end
+
+                              # Returns an InstrumentationLibrary struct for backwards compatibility.
+                              # @deprecated Please use instrumentation_scope instead.
+                              #
+                              # @return InstrumentationLibrary
+                              def instrumentation_library
+                                @instrumentation_library ||= InstrumentationLibrary.new(
+                                  @instrumentation_scope.name,
+                                  @instrumentation_scope.version
+                                )
+                              end
```

Comment:
I would instead rely on duck-typing and return `@instrumentation_scope` here. I consider the documented types to be informative ("what's the shape of this thing?") rather than strict - no published exporter should care about the type returned from `instrumentation_library`, only that it has a `name` and a `version`.
```suggestion
                              alias instrumentation_library instrumentation_scope
```

On the other hand, we *do* need to retain the `InstrumentationLibrary` type so that anyone explicitly building a `SpanData` for testing doesn't need to update their test code (yet).

Reply:
I was on the fence about that myself; thanks for pushing me to a side! I'll update throughout. 😄